text/0000-pub-sub.md

```diff
@@ -0,0 +1,171 @@
+  - `GET /pubsub/history`
+  - `GET /pubsub/history?corellationId={corellationId}`
+
+It is allowed for publishers to provide callback URLs to receive feedback from `mod-pubsub` module in cases when:
+- Subscriber did not receive an event within the specified period of time.
+- All subscribers deleted their subscriptions on a particular event type registered by this publisher.
+ 
+
+There is a set of requirements that must be followed by a module to provide “Subscriber endpoints” to consume events. The basic set of requirements:
+- HTTP Method - POST
+- A set of required HTTP header. [_To be defined further if it is needed._]
+- A payload structure - JSON which represents an event. See [Event data structure](#Event data structure) 
+
+Reference implementations should be provided that allow a back-end module to become a publisher or a subscriber respectively to simplify development.
+  
+A dedicated module, `mod-pubsub`, must implement and encapsulate all logic related to event sending and delivery. All interactions between this module and back-end business modules will be done via HTTP.
+
+#### Registration
+
+Each module which acts as a publisher or a subscriber must contain a configuration or property with a set of event types it deals with. Publishers should provide event descriptors. For subscribers, it is enough to specify event types together with endpoints for event delivery. Details of that configuration must be defined further.
+
+##### Event Type registration 
+The set of event descriptors will be registered by a publisher at the time when the module is enabled for a tenant. To do that the publisher must call `pubsub` API for event type registration. The payload for that call is a set of event descriptors which define events this particular module is able to publish.
+
+At this stage it is possible to check if another module has already been registered as a publisher for the same event descriptors.
+
+It is worth mentioning that at this stage only event descriptors will be registered. To allow a module to publish events further steps optionally should be taken.
+
+![Event type registration diagram](http://www.plantuml.com/plantuml/png/TP4nJyCm48Lt_ufJ9nWgTKPL9nZO4B01myM-D2SSsxBF5VttN1nQXL8YPOZFxxltNeJ54pmSA0Z8CL61J7ikab4u1cEmIWnrx2Z3QjljtZQRpSVAc8HPs792bQt6wDRRL3lFPQ2yMm4MICROw3tP2LFcEtgbrAwDoP4nwnsaEq3GMLKJhCS1Eq_kFVGnwq3qEuWn-nnhMfZyxH7qIJ33dWcT2Wi9n1weAHmLcpb9p4N1EOvwmSRGyxvU4jJMn9ZW6dXU-DfQkVaFUTlKAZphdp0NkhORWmCxaFs45izdDNlyoKAPt79Xew8Nu1uSzxtETVG7saWo41GSJpjRG_Ibq0PPsDE27iC1FPzMtPz9SmglnaDixNYsCk5hS9UObl95HLTImXJIS_W4YStO_7ElE5csE-21e2CAYpPcHxfzUhy1)
+
+##### Subscriber registration
+A back-end module can register event types it can consume at the time when the module is enabled for a tenant. To do that it must call the `pubsub` API for registration. The payload for that call is a set of event types augmented with endpoints provided by this module. These endpoints will be used to deliver events of respective types.
```

Comment:
What should happen if registration fails? Should tenant activation for that module fail?

Reply:
See above.

Reply:
Thanks